build-tools/xaprepare/xaprepare/Application/Utilities.cs

```diff
@@ -471,6 +417,27 @@ public static async Task<(bool success, ulong size, HttpStatusCode status)> GetD
 			return (false, 0, HttpStatusCode.InternalServerError);
 		}
 
+		public static async Task<string> GetUrlContent (Uri url)
```

Comment:
I wonder if it would make sense to cache the downloaded content on disk and throw the exception only if there's no cached content and it cannot be fetched.  This would make our build process (slightly) more offline friendly.

Reply:
I like this idea, or alternatively we could copy the license content directly like we've done for other TPN items? This also brings up a broader question I had about the TPN removals in this PR. I'm not entirely sure what mono related licenses we should still include in our TPN now that we are redistributing fewer upstream items. @grendello do you have any thoughts on that?

Reply:
I think downloading licenses is fine, no reason to put extra stuff in our repo.  With regards to TPN removal, it feels to me that with this PR we could probably drop almost all of the Mono licenses, since there's not much left that we directly use. What's left is Cecil, isn't it?

Reply:
To simplify I decided to strip this out, and instead import the license content for Cecil and Linker (due to some linker sources that were recently included in commit https://github.com/xamarin/xamarin-android/commit/5daf1027666098ef43e21d7692daf1513024819e.